### DISCUSSION TITLE
Manual maintenance of platform directory?

### DISCUSSION THREAD

**Discussion by passwhl** [ionic-team/capacitor#6650] [General]
Why do we need to maintain the configuration files under the platform directory if hybrid development frameworks are supposed to avoid maintaining native code?

**Comment by jcesarmobile** [comment]
> if hybrid development frameworks are supposed to avoid maintaining native code

why are they supposed to do that?

Anyway, it's not mandatory to make config changes in the native platforms and commit them, it's just recommended.
We also offer [trapeze](https://trapeze.dev/) to help you modify your native project with config files or programmatically if you want to follow that route instead of committing the native projects.

**Comment by mlynch** [comment]
That's not the point of hybrid frameworks, in our opinion. The point is to enable you to focus on a 95%+ cross-platform, standard web codebase, but be able to do anything natively you might need to.

Capacitor makes it extremely easy to quickly customize or extend the native app your web app is running in, and we believe that's the best trade-off. Compared to Cordova where there are layers of abstraction between you and the native project which you are inevitably going to want to modify or extend. In my experience that created a "Cordova way" to do anything and tooling that would constantly run into issues and produce developer frustration.

This is a deliberate design decision and we believe it's the right one because you never get stuck and we can keep tooling a lot simpler and less prone to breaking.
